src/App.tsx

```diff
@@ -1,61 +1,134 @@
 import './App.scss';
+import React, { useState } from 'react';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const initialTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(initialTodos);
+  const [newTitle, setTitle] = useState('');
+  const [selectedUser, setselectedUser] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleAddTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const noTitle: boolean = !newTitle || newTitle === ' ';
```

Comment:
No need to write primitive types.
```suggestion
    const noTitle = !newTitle || newTitle === ' ';
```